src/main/java/com/streamr/client/utils/MessageCreationUtil.java

```diff
@@ -179,7 +228,7 @@ private int getStreamPartition(int nbPartitions, String partitionKey) {
     private Pair<MessageID, MessageRef> createMsgIdAndRef(String streamId, int streamPartition, long timestamp) {
         String key = streamId + streamPartition;
         long sequenceNumber = getNextSequenceNumber(key, timestamp);
-        MessageID msgId = new MessageID(streamId, streamPartition, timestamp, sequenceNumber, publisherId, msgChainId);
+        MessageID msgId = new MessageID(streamId, streamPartition, timestamp, sequenceNumber, publisherId.toString(), msgChainId);
```

Comment:
perhaps `MessageID` should take a `publisherId` as `Address` instead of `string`? or too much refactoring?

Reply:
Yeah, I couldn't bring myself to doing that refactor on the same sitting. I'll do it!